src/rgw/rgw_rest_s3.h

```diff
@@ -914,42 +914,129 @@ class s3select;
 class csv_object;
 }
 
+
+class aws_response_handler
+{//TODO this class should reside on s3select submodule 
```

Comment:
what's the plan for this part?

if we're going to keep this in rgw, i'd really prefer we move it to separate files like `rgw_rest_s3select.*`. rgw_rest_s3.cc is already huge

Reply:
agreed.
rgw_rest_s3.[cc,h] is too big.

generally, it's better to move any functionality which is not RGW-related to s3select, in the case of `aws_response_handler` it needs some more refactoring.